cirrus/google_cloud/manager.py

```diff
@@ -917,6 +934,94 @@ def delete_service_account_key(self, account, key_name):
 
         return response.json()
 
+    def give_service_account_billing_access(
+        self, account_id, project_id=None, billing_role_name=None
+    ):
+        """
+        Give the specified service account id permissions to bill to the provided project
+        id. Project id will default to the current project.
+
+        Args:
+            account_id (str): email address or the uniqueId of the service account
+            project_id (str): Google project identifier
+        """
+        # default to setting billing rights to the current project
+        project_id = project_id or self.project_id
+        project_resource = "projects/" + project_id
+        billing_role_id = billing_role_name or "custom.ProjectBillingUser"
+        full_billing_role_resource = "projects/{}/roles/{}".format(
+            project_id, billing_role_id
+        )
+
+        # get project IAM policy and see if the sa already has the necessary access
+        policy = self._get_project_iam_policy(project_id)
+        role = GooglePolicyRole(name=full_billing_role_resource)
+        roles = {role.name: role for role in policy.roles}
+        if full_billing_role_resource in roles.keys():
+            role = roles[full_billing_role_resource]
+            if account_id in [member.email_id for member in role.members]:
+                logger.info(
+                    "Custom role {} already exists on project {} for service account {}".format(
+                        billing_role_id, project_id, account_id
+                    )
+                )
+                return
+
+        # create new role with just billing access if doesn't exist already
+        # https://cloud.google.com/iam/docs/creating-custom-roles#iam-custom-roles-create-rest
+        new_role = {
+            "role_id": billing_role_id,
+            "role": {
+                "name": "",
+                "title": "Project Billing User",
+                "description": "This role grants a user access to use billing for the given resource.",
+                "included_permissions": "serviceusage.services.use",
```

Comment:
how did you choose `serviceusage.services.use`? i don't see in the doc that it gives permission to bill the project and nothing else

Reply:
https://cloud.google.com/storage/docs/access-control/iam-console
`All actions that include a billing project in the request require serviceusage.services.use permission for the project that's specified.`